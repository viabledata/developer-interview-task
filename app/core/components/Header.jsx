import PropTypes from 'prop-types';
import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// local imports
import AppConstants from "../../common/AppConstants";
import config from '../../config/core';
import secureLocalStorage from '../../common/security/SecureLocalStorage';

const { serviceDesk } = config;

export class Header extends React.Component {
    constructor(props) {
        super(props);
        this.secureLocalStorage = secureLocalStorage;
        this.dashboard = this.dashboard.bind(this);
    }

    dashboard(event) {
        event.preventDefault();
        this.props.history.push(AppConstants.DASHBOARD_PATH);
    }

    render() {
        return <header className="govuk-header" role="banner" data-module="header">
                <div className="govuk-header__container govuk-width-container">
                    <div className="govuk-header__content" style={{width: '100%'}}>
                        <div className="govuk-grid-row">
                            <div className="govuk-grid-column-one-half">
                                <a
                                    id="dashboard"
                                    href={AppConstants.DASHBOARD_PATH}
                                    onClick={event => this.dashboard(event)}
                                    className="govuk-header__link govuk-header__link--service-name"
                                >Central Operations Platform</a>
                            </div>
                            <div className="govuk-grid-column-one-half header-nav">
                                <a
                                    id="profile"
                                    href="#"
                                    className="govuk-header__link header-nav__link"
                                >My profile</a>
                                <a
                                    id="support"
                                    className="govuk-header__link header-nav__link"
                                    href={`${serviceDesk.support}`}
                                    target="_blank"
                                    rel="noopener noreferrer"
                                >Support</a>
                                <a
                                    id="logout"
                                    href="#"
                                    className="govuk-header__link header-nav__link"
                                >Sign out</a>
                            </div>
                        </div>
                    </div>
                </div>
            </header>;
    }
}

Header.propTypes = {
    history: PropTypes.shape({
        push: PropTypes.func,
    }).isRequired,
};

const mapDispatchToProps = dispatch => bindActionCreators({}, dispatch);

export default withRouter(connect(state => ({
    appConfig: state.appConfig,
}), mapDispatchToProps)(Header));
