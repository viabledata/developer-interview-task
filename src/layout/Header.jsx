import Nav from './Nav';

const Header = () => (
  <header className="thisApp-header" role="banner" data-module="thisApp-header">
    <a href="#content" className="thisApp-skip-link">Skip to main content</a>
    <div className="thisApp-header__container thisApp-width-container">

      <Nav />
    </div>
  </header>
);
export default Header;
