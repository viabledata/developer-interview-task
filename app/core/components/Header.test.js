import React from 'react';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history'
import { Header } from './Header';

describe('Header', () => {

  const props = {
    history: createMemoryHistory(),
    kc: {
      logout: jest.fn(),
    },
  }

  it('renders without crashing', () => {
    const wrapper = shallow(<Header { ...props }/>);
    expect(wrapper.exists()).toBe(true);
  });

  it('renders Profile link', () => {
    const wrapper = shallow(<Header { ...props }/>);
    const node = wrapper.find('#profile');
    expect(node.text()).toBe('My profile');
    expect(node.prop('href')).toBe('/submit-a-form/edit-your-profile');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Header { ...props }/>);
    expect(wrapper).toMatchSnapshot();
  });

});
