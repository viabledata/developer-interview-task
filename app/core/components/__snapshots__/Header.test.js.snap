// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="govuk-header"
  data-module="header"
  role="banner"
>
  <div
    className="govuk-header__container govuk-width-container"
  >
    <div
      className="govuk-header__content"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-one-half"
        >
          <a
            className="govuk-header__link govuk-header__link--service-name"
            href="/dashboard"
            id="dashboard"
            onClick={[Function]}
          >
            Central Operations Platform
          </a>
        </div>
        <div
          className="govuk-grid-column-one-half header-nav"
        >
          <a
            className="govuk-header__link header-nav__link"
            href="/submit-a-form/edit-your-profile"
            id="profile"
            onClick={[Function]}
          >
            My profile
          </a>
          <a
            className="govuk-header__link header-nav__link"
            href="https://support.cop.homeoffice.gov.uk/servicedesk/customer/portal/3"
            id="support"
            rel="noopener noreferrer"
            target="_blank"
          >
            Support
          </a>
          <a
            className="govuk-header__link header-nav__link"
            href="/logout"
            id="logout"
            onClick={[Function]}
          >
            Sign out
          </a>
        </div>
      </div>
    </div>
  </div>
</header>
`;
